File: Aquecimento/palindromoC/palindromo.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>
#define MAX 100

void imprimir(bool isPalindromo){
	if(isPalindromo){
		printf("SIM\n");
	} else {
		printf("NAO\n");
	}
	
}

bool isPalindromo(char pal[]){
	int count = 0, tamString = strlen(pal);
	
	for(int i = 0; i < tamString/2; i++){
		if(pal[i] != pal[tamString-1-i]){
			return false;
		}
	}
	return true;
}

int main(){
	char string[MAX];
	while(1){
		scanf(" %[^\n]", string);
		
		if(strcmp(string, "FIM") == 0){
			break;
		} 
		bool isPal = isPalindromo(string);
		imprimir(isPal);
	}
	
	return ( 0 );
}

File: Aquecimento/palindromoJava/palindromo.java

class palindromo {
    public static void main(String[] args){
        
        String string ;
        
        while(!(string = MyIO.readString()).equals("FIM")){
            boolean isPalindromo = isPalindromo(string);

            imprimir(isPalindromo);
        }
    }

    public static boolean isPalindromo(String s){
        int tamString = s.length()-1, aux = 0;

        for(int i = 0, j = s.length()-1; i < s.length(); i++){
            if(s.charAt(i) == s.charAt(j)) {
                aux++;
            }
            j--;
        }
        if(aux == tamString/2) {
            return true;
        } 
        return false;
        
    }

    public static void imprimir(boolean bool){
        if(bool){
            MyIO.println("SIM");
        } else {
            MyIO.println("NAO");
        }
    }
}